### PR TITLE
Impl std::error::Error for Error type

### PR DESCRIPTION
**Pull request by cmbrandenburg**

### CODE REVIEW

File: src/enums.rs [hunk 1]
```diff
@@ -10,6 +11,24 @@ pub enum Error {
...
 
+impl std::fmt::Display for Error {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> Result<(), std::fmt::Error> {
+        (self as &std::error::Error).description().fmt(f)
+    }
+}
+
+impl std::error::Error for Error {
+    fn description(&self) -> &str {
+        match self {
```
Comment (GuillaumeGomez) [inline src/enums.rs]: Generally we match over `*self` to avoid having to put `&` in front of every pattern matching.

**Comment by GuillaumeGomez** [comment]
Just a small nit to fix. Otherwise it's really nice, thanks!

**Comment by cmbrandenburg** [comment]
Fixed.

Please, will you publish a new release on crates.io including this change? I would like to use this crate in one of my projects, and I would like to have this feature without sourcing a forked repo.

**Comment by GuillaumeGomez** [comment]
Sure. Thanks!

**Pull request merged**